electrum/transaction.py

```diff
@@ -1851,12 +1851,17 @@ def serialize_preimage(self, txin_index: int, *,
         inputs = self.inputs()
         outputs = self.outputs()
         txin = inputs[txin_index]
-        sighash = txin.sighash if txin.sighash is not None else SIGHASH_ALL
-        if sighash != SIGHASH_ALL:
-            raise Exception("only SIGHASH_ALL signing is supported!")
+        if txin.sighash is None:
+            txin.sighash = 0
+            # raise Exception("SigHash of txin couldn't be None!")
+        sighash = txin.sighash | SIGHASH_ALL | constants.net.SIGHASH_FORK_BTG
```

Comment:
Let's switch back to the original impl, and always add `constants.net.SIGHASH_FORK_BTG` to the sighash flag